app/src/lib/git/rebase.ts

```diff
@@ -82,6 +86,73 @@ export async function getRebaseContext(
 
   return null
 }
+
+/** Regex for identifying when rebase applied each commit onto the base branch */
+const rebaseApplyingRe = /^Applying: (.*)/
+
+/**
+ * A parser to read and emit rebase progress from Git `stdout`
+ */
+class GitRebaseParser {
+  private currentCommitCount = 0
+
+  public constructor(
+    startCount: number,
+    private readonly totalCommitCount: number
+  ) {
+    this.currentCommitCount = startCount
+  }
+
+  public parse(line: string): IRebaseProgress | null {
+    const match = rebaseApplyingRe.exec(line)
+    if (match == null || match.length !== 2) {
+      // Git will sometimes emit other output (for example, when it tries to
+      // resolve conflicts) and this does not match the expected output
+      return null
+    }
+
+    const commitSummary = match[1]
+    this.currentCommitCount++
+
+    const value = this.currentCommitCount / this.totalCommitCount
+
+    return {
+      kind: 'rebase',
+      title: `Rebasing ${this.currentCommitCount} of ${
```

Comment:
Assuming this is something that'll be presented to the user could we add a word about what the unit of work is here?

```suggestion
      title: `Rebasing commit ${this.currentCommitCount} of ${
```

Reply:
👍 - current designs can be found here https://github.com/desktop/desktop/issues/6962